### PR TITLE
Fix removed service-to-service peering links

### PR DESCRIPTION
**Pull request by karras**
### Description

In the PR #16291 the Mesh Gateway service-to-service peering page was removed (`website/content/docs/connect/gateways/mesh-gateway/service-to-service-traffic-peers.mdx`). Though it looks like not all links to that page were replaced. This PR tries to fix those occurrences.

I tried to find the best matching replacements to capture the intentions of the affected sections. Please feel free to suggest better alternatives and your raise concerns where you see fit. :)

### Testing & Reproduction steps

1. Navigate to https://developer.hashicorp.com/consul/docs/connect/gateways/mesh-gateway/service-to-service-traffic-peers
2. 404 is presented

### Links

Affected pages:

* https://developer.hashicorp.com/consul/docs/connect/gateways#mesh-gateways
* https://developer.hashicorp.com/consul/docs/connect/gateways/mesh-gateway
* https://developer.hashicorp.com/consul/docs/connect/gateways/mesh-gateway/peering-via-mesh-gateways
* https://developer.hashicorp.com/consul/docs/enterprise/admin-partitions
* https://developer.hashicorp.com/consul/docs/lambda/invoke-from-lambda 

### PR Checklist

* [x] updated test coverage (n/a)
* [ ] external facing docs updated
* [ ] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

File: website/content/docs/connect/gateways/mesh-gateway/peering-via-mesh-gateways.mdx [hunk 1]
```diff
@@ -7,7 +7,7 @@ description: >-
 
 # Enabling Peering Control Plane Traffic
 
-In addition to [service-to-service traffic routing](/consul/docs/connect/gateways/mesh-gateway/service-to-service-traffic-peers),
+In addition to [service-to-service traffic routing](/consul/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions),
```
Comment (trujillo-adam) [inline website/content/docs/connect/gateways/mesh-gateway/peering-via-mesh-gateways.mdx:10]:
```suggestion
This topic describes how to configure a mesh gateway to route control plane traffic between Consul clusters that share a peer connection. For information about routing service traffic between cluster peers through a mesh gateway, refer to [Enabling Service-to-service Traffic Across Admin Partitions](/consul/docs/connect/gateways/mesh-gateway/service-to-service-traffic-partitions).
```
I know that it's not in the scope of this PR, but we should take this opportunity while editing this page to add some clarity about the what this page is about.

**Review by trujillo-adam — APPROVED** [review]
Thanks for fixing these links. I added one suggestion, but pre-approving so that you aren't blocked.

File: website/content/docs/connect/gateways/mesh-gateway/peering-via-mesh-gateways.mdx (hunk 1 above)
Comment (karras) [reply website/content/docs/connect/gateways/mesh-gateway/peering-via-mesh-gateways.mdx]: Thanks a lot for the suggestion, I've extended the section accordingly.

**Comment by karras** [comment]
@trujillo-adam Should be ready to be merged, what do you think?

**Comment by karras** [comment]
@trujillo-adam What's the status here? I guess this PR starts to become obsolete with more and more changes being applied to the docs, which is a bummer considering the invested effort.

**Comment by trujillo-adam** [comment]
> @trujillo-adam What's the status here? I guess this PR starts to become obsolete with more and more changes being applied to the docs, which is a bummer considering the invested effort.

Ah, apologies, I didn't realize that you weren't able to merge. I'll push through.

**Review by trujillo-adam — APPROVED** [review]

**Pull request merged**